clients/src/main/java/org/apache/kafka/common/config/ConfigDef.java

```diff
@@ -76,7 +76,7 @@
      * A unique Java object which represents the lack of a default value.<p>
      * The 'new' here is intentional.
      */
-    public static final Object NO_DEFAULT_VALUE = new String("");
+    public static final Object NO_DEFAULT_VALUE = new Object();
```

Comment:
I am OK with this and I suggested that when @cmccabe got confused by a findBugs warning suggesting that the `new` was unnecessary. If we do this, we should remove the `The new here is intentional` from the comment above since there's no alternative when using `Object`.

Having said that, I don't think we should change `==` to `equals`. The usage of `==` makes it clear that we are using reference equality and it's the common pattern when using sentinels in Java.